### DISCUSSION TITLE
Adding Metrics from within the Application

### DISCUSSION THREAD

**Discussion by Taragos** [open-telemetry/opentelemetry-java-instrumentation#5184] [Q&A] [answered]
Hello,

I am using the Opentelemetry-Javaagent to observe a Karaf-based application.  
The collection/exporting of the traces works perfectly and I can also access the Spans from within the application to add attributes or retrieve the current traceId to add it to the logs.  

I now wanted to add some application-specific metrics, for example how often a certain file was openend etc. and export them via the opentelemetry-javaagent.  

I tried to access the javaagents-meter with the following code, but failed to add metrics to it (they didn't show up)

```
    private static final MeterProvider meterProvider = MeterProvider.noop();
    private static final Meter meter = meterProvider.get("io.opentelemetry.javaagent.opentelemetry-javaagent");
    private static final LongCounter counter = meter.counterBuilder("status_calls").setDescription("Number of Times Status was called").setUnit("1").build();

    ....

    counter.add(1)
```

Javaagent Version: 1.10.0
Opentelemetry API Version: 1.10.0

Is this possible? If yes, how could I implement something like this this?

**Comment by laurit** [comment, marked as answer]
Instead of `MeterProvider.noop()` try with `GlobalOpenTelemetry.getMeterProvider()`

**Reply by Taragos** [reply]
Worked like a charm, thanks!
